direct/tpm2/policy_test.go

```diff
@@ -314,42 +324,32 @@ func TestPolicyPCRUpdate(t *testing.T) {
 		},
 	}
 
-	pcrRead := PCRRead{
-		PCRSelectionIn: selection,
-	}
-
-	pcrReadRsp, err := pcrRead.Execute(thetpm)
-	if err != nil {
-		t.Fatalf("Failed to read PCRSelection")
-	}
-
-	var expectedVal []byte
-	for _, digest := range pcrReadRsp.PCRValues.Digests {
-		expectedVal = append(expectedVal, digest.Buffer...)
-	}
-	t.Logf("expectedVal=%x", expectedVal)
-
-	// Hash algorithm must match the one in PolicySession.
-	expectedDigest := sha1.Sum(expectedVal)
+	expectedDigest := getExpectedPCRDigest(t, thetpm, selection, tpm.AlgSHA1)
 	t.Logf("expectedDigest=%x", expectedDigest)
 
-	wrongCase := append(expectedVal, []byte("wrongCase")...)
-	wrongDigest := sha1.Sum(wrongCase)
+	wrongDigest := sha1.Sum(expectedDigest[:])
+	t.Logf("wrongDigest=%x", wrongDigest)
 
 	tests := []struct {
 		desc           string
 		expectedDigest []byte
 		expectedResult bool
```

Comment:
I'm not really sure what this expected result is? When it is false, the test code seems to check that changing the policyPCR's PCRDigest field causes the policy not to match, but that's not as interesting as making sure that the TPM and the PolicyCalculator always agree on what a policy hash is, for a given policy. IMHO, I think you could take out the failure case and make this a flat, one-test test.

Reply:
After discussing with Joe, I decided to rewrite the whole test covering 6 cases, Trial correct, incorrect, empty, Real correct, incorrect, empty.